build.sc

```diff
@@ -0,0 +1,94 @@
+class HamsterModule(val scalaCrossVersion: String) extends Module {
+  override def millSourcePath = super.millSourcePath / up / up
+
+  trait HamsterModule extends SbtModule {
+
+    def scalaVersion = scalaCrossVersion
+
+    override def scalacOptions = Seq("-Xplugin-require:macroparadise", "-language:higherKinds", "-language:implicitConversions", "-feature")
+
+    def publishVersion = "2.6.0"
+
+    def pomSettings = PomSettings(
+      description = "A mini Scala utility library",
+      organization = "io.github.scala-hamsters",
+      url = "https://github.com/scala-hamsters/hamsters",
+      licenses = Seq(License.`Apache-2.0`),
+      versionControl = VersionControl.github("scala-hamsters", "hamsters"),
+      developers = Seq(
+        Developer("loicdescotte", "Loïc Descotte","http://loicdescotte.github.io"),
+        Developer("dgouyette", "Damien Gouyette","http://www.Cestpasdur.com"),
+        Developer("oraclewalid", "Walid Chergui","https://github.com/oraclewalid")
+      )
+    )
+  }
+
+  object jvm extends WithMacroParadise {
+    override def moduleDeps = Seq(shared)
+    object test extends Tests {
+      override def scalaVersion = "2.12.4"
+      override def ivyDeps = Agg(
+        ivy"org.scalatest::scalatest:3.0.1",
+        ivy"org.scalamock::scalamock-scalatest-support:3.6.0",
+        ivy"org.scalacheck::scalacheck:1.13.4",
+        ivy"ca.mrvisser::sealerate:0.0.5"
+      )
+      def testFrameworks = Seq("org.scalatest.tools.Framework")
+    }
+  }
+  object macros extends WithMacroParadise {
+    override def millSourcePath = super.millSourcePath / 'shared
+  }
+  object metas extends WithMetaParadise {
+    override def millSourcePath = super.millSourcePath / 'shared
+  }
+  object shared extends WithMetaParadise {
+    override def moduleDeps = Seq(macros, metas)
+  }
+
+  trait WithMetaParadise extends HamsterModule {
+    override def repositories() = super.repositories ++ Seq(
+      coursier.ivy.IvyRepository("https://dl.bintray.com/scalameta/maven/", dropInfoAttributes = true)
+    )
+    override def ivyDeps = Agg(
+      ivy"org.scalameta::scalameta:1.8.0"
+    )
+    override def scalacPluginIvyDeps = super.scalacPluginIvyDeps() ++ Agg(
+      ivy"org.scalameta:::paradise:3.0.0-M10"
+    )
+  }
+
+  trait WithMacroParadise extends HamsterModule {
+    override def ivyDeps = Agg(
+      ivy"org.scala-lang:scala-reflect:$scalaCrossVersion"
+    )
+    override def scalacPluginIvyDeps = super.scalacPluginIvyDeps() ++ Agg(
+      ivy"org.scalamacros:::paradise:2.1.0"
+    )
+  }
+}
+
+object js extends ScalaJSModule {
+  def scalaVersion = "2.12.4"
+  def scalaJSVersion = "0.6.22"
+
+  override def sources = T.sources(
+    millSourcePath / 'shared / 'src / 'main,
+    millSourcePath / 'metas / 'src / 'main,
+    millSourcePath / 'macros / 'src / 'main
+  )
+}
+
+object release extends Module {
+  def updateReadme(releaseVersion: String, documentationVersion: String)= T.command{
```

Comment:
You can run this task with: `mill release.updateReadme --releaseVersion 2.7.0 --documentationVersion 2.7` or call it in the build process